app/lib/square_checkout/client.rb

```diff
@@ -0,0 +1,109 @@
+module SquareCheckout
+  # SquareCheckout::Client wraps the Square::Client and exposes a few specific intereactions used
+  # by Circulate.
+  #
+  # In the code, if the namespace is Square::, it is a part of the square.rb Rubygem; if it is
+  # SquareCheckout::, that's ours.
+  class Client
+    def initialize(access_token:, location_id:, environment: "production", now: Time.current)
+      bearer_auth_credential = Square::BearerAuthCredentials.new(access_token: access_token)
+      @client = Square::Client.new(bearer_auth_credentials: bearer_auth_credential, environment: environment)
+      @location_id = location_id
+      @now = now
+    end
+
+    def checkout_url(amount:, email:, date:, member_id:, return_to:, idempotency_key: random_idempotency_key)
+      checkout_response = @client.checkout.create_payment_link(
+        body: {
+          idempotency_key: idempotency_key,
+          pre_populated_data: {
+            buyer_email: email
+          },
+          checkout_options: {
+            redirect_url: return_to,
+            allow_tipping: false,
+            enable_coupon: false,
+            enable_loyalty: false,
+            accepted_payment_methods: {
+              apple_pay: true,
+              google_pay: true,
+              cash_app_pay: true
+            }
+          },
+          order: {
+            reference_id: "#{date.strftime("%Y%m%d")}-#{member_id}",
+            location_id: @location_id,
+
+            line_items: [{
+              name: "Annual Membership",
+              quantity: "1",
+              base_price_money: {
+                amount: amount.cents,
+                currency: "USD"
+              }
+            }],
+
+            metadata: {
+              member_id: member_id.to_s,
+              created_by: "circulate"
+            }
```

Comment:
All payments (including those made using the POS system to sell shirts) will send us web hooks, so we use this metadata to detect if payments were created by our app. If they weren't, we ignore them in the webhook handler.